persistence-modules/spring-data-jpa-query-4/src/main/java/com/baeldung/spring/data/jpa/maxvalue/EmployeeRepository.java

```diff
@@ -0,0 +1,34 @@
+package com.baeldung.spring.data.jpa.maxvalue;
+
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.Comparator;
+import java.util.Optional;
+
+public interface EmployeeRepository extends JpaRepository<Employee, Integer> {
+    Optional<Employee> findTopByOrderBySalaryDesc();
+
+    Optional<EmployeeSalary> findTopSalaryByOrderBySalaryDesc();
+
+    @Query("SELECT MAX(e.salary) FROM Employee e")
+    Optional<Long> findTopSalaryJQPL();
```

Comment:
We follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case for naming conventions and so this should be Jqpl not JQPL